### PR TITLE
Remove check that throws when HiveSplitSource buffers too much

### PR DESCRIPTION
**Pull request by haozhun**

### CODE REVIEW

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java [hunk 1]
```diff
@@ -82,6 +85,7 @@
     private final int maxPartitionBatchSize;
     private final int maxInitialSplits;
     private final boolean recursiveDfsWalkerEnabled;
+    private final CounterStat highMemorySplitSourceCounter;
```
Comment (electrum) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java]: Why not initialize this here?

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitSource.java [hunk 2]
```diff
@@ -100,6 +114,12 @@ int getOutstandingSplitCount()
                 // TODO: throw an exception when this is hit.
                 // For bucketed tables, push back mechanism is not respected when building splits for each partition.
                 // For such tables, the throw here could fire and fail the query.
+                if (loggedHighMemoryWarning.compareAndSet(false, true)) {
+                    highMemorySplitSourceCounter.update(1);
+                    log.warn(
+                            "Split buffering for %s.%s in query %s exceeded memory limit (%s). %s splits are buffered.",
```
Comment (electrum) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitSource.java]: Keep this on the previous line

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java (hunk 1 above)
Comment (haozhun) [reply presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java]: Both constructors in HiveSplitManager are public. When the second one is used, this counter is passed in from outside.

**Comment by haozhun** [comment]
Merged #9175

**Pull request closed without merging**